docs/api/middleware.rst

```diff
@@ -26,8 +26,11 @@ defined below.
 
         .. code:: python
 
+            from typing import Any
+            from falcon import Request, Response
+
             class ExampleMiddleware:
-                def process_request(self, req, resp):
+                def process_request(self, req: Request, resp: Response) -> None:
```

Comment:
I'm not 100% convinced we should type middleware definitions... :thinking:
It makes things clearer, but OTOH, the definitions get a bit dense.

Reply:
I think is makes sense, since these are docs and typing is helpful also as documentation.